File: src/main/java/com/hrms/hr_payroll_management_system/service/report/ReportService.java
package com.hrms.hr_payroll_management_system.service.report;

import com.hrms.hr_payroll_management_system.dto.response.report.AttendanceReportResponse;
import com.hrms.hr_payroll_management_system.dto.response.report.EmployeeReportResponse;
import com.hrms.hr_payroll_management_system.dto.response.report.LeaveReportResponse;
import com.hrms.hr_payroll_management_system.dto.response.report.PayrollReportResponse;

import java.time.LocalDate;
import java.util.List;

public interface ReportService {

    List<EmployeeReportResponse> getEmployeeReport(Long departmentId);

    List<AttendanceReportResponse> getAttendanceReport(
            LocalDate startDate,
            LocalDate endDate
    );

    List<LeaveReportResponse> getLeaveReport(Integer year);

    List<PayrollReportResponse> getPayrollReport(
            Integer year,
            Integer month
    );
}

File: src/services/report.service.js
import { reportApi } from '../api/report.api';

// ⚠️ Report endpoints return the List directly (no ApiResponse wrapper) —
// axiosClient interceptor already unwraps response.data, so `res` here
// IS the array itself, not `res.data`.

export const reportService = {
  async employees(departmentId) {
    const res = await reportApi.getEmployeeReport(departmentId || undefined);
    return res || [];
  },
  async attendance(startDate, endDate) {
    const res = await reportApi.getAttendanceReport(startDate, endDate);
    return res || [];
  },
  async leaves(year) {
    const res = await reportApi.getLeaveReport(year);
    return res || [];
  },
  async payroll(year, month) {
    const res = await reportApi.getPayrollReport(year, month);
    return res || [];
  },
};